### ISSUE TITLE
gpu accelerated renderer

### ISSUE THREAD

**Issue by joeblew99** [Yeicor/sdfx-ui#4]
https://github.com/polyred/polyred works with stl files.

I tested polyred and its reasonably fast. 
It supported desktops and mobiles using opengl and metal renderers
It has orbit tool integrated too.
Its using go 1.18 and some generics also.

I should be quite easy to integrate with sfx.ui.

**Comment by yeicor** [comment]
The main idea of SDFX-UI is quickly exploring the SDF without having to construct the mesh first, for fast(er) development iterations. Hence, the current main 3D renderer is the raycasting renderer which has no load time, and the mesh-based renderer is only included as a fallback.

I see no performance problem with the current mesh-based renderer. I generated a mesh with 182k triangles which already would add seconds to load times, and it is capable of rendering it at a resolution of 640x480 at 30FPS with some small lag spikes. This tool is not intended for final high-resolution and high-performance renders, just quick previews. Furthermore, [polyred](https://github.com/polyred/polyred) implements its own low-level graphics, which would make it incompatible with [ebiten](https://ebiten.org/). Maybe a mesh renderer could be implemented using [ebiten's shaders](https://ebiten.org/documents/shader.html) instead, but that is not a priority.

I have just implemented #5, a new option which renders the scene while moving the camera (for 2D and both 3D renderers) to show this and improve usability of this tool. The following is the previously mentioned demo, running realtime:

![demo](https://user-images.githubusercontent.com/4929005/160677170-b3255dbe-db37-49af-a2a3-39862ef70c69.gif)
